### PR TITLE
bpo-33649: Add a high-level section about Futures; few quick fixes

### PR DESCRIPTION
**Pull request by 1st1**
https://bugs.python.org/issue33649

### CODE REVIEW

**Review by elprans — APPROVED** [review]

File: Doc/library/asyncio-task.rst [hunk 1]
```diff
@@ -112,14 +112,49 @@ two closely related concepts:
...
+    async def main():
+        await function_that_returns_a_future_object()
+
+        # this is also valid:
+        await asyncio.gather(
+            function_that_returns_a_future_object(),
+            some_python_coroutine()
+        )
+
+asyncio :class:`Tasks <Task>` have compatible API with Futures
```
Comment (willingc) [inline Doc/library/asyncio-task.rst]: ..have a compatible API...

File: Doc/library/asyncio-task.rst [hunk 2]
```diff
@@ -112,14 +112,49 @@ two closely related concepts:
...
+with regards to cancellation and introspection.
+
+Many asyncio APIs are documented to accept coroutines, Tasks, or
+Futures.  What it means is that they are ready to accept anything
```
Comment (willingc) [inline Doc/library/asyncio-task.rst]: By doing so, the async APIs are ready to accept anything which you can await on.

File: Doc/library/asyncio-task.rst [hunk 3]
```diff
@@ -191,30 +238,32 @@ Running Tasks Concurrently
 
 .. function:: gather(\*fs, loop=None, return_exceptions=False)
 
-   Return a Future aggregating results from the given coroutine objects,
-   Tasks, or Futures.
+   Return a :class:`Future` aggregating results from the given
```
Comment (willingc) [inline Doc/library/asyncio-task.rst]: *fs is used in this section without any prior mention on the page. We should have a brief verbal description here.

File: Doc/library/asyncio-task.rst [hunk 4]
```diff
@@ -235,6 +284,7 @@ Running Tasks Concurrently
           print(f"Task {name}: factorial({number}) = {f}")
 
       async def main():
+          # Schedule three calls *concurrently*:
```
Comment (willingc) [inline Doc/library/asyncio-task.rst:303]: Nice!

**Review by willingc — APPROVED** [review]
Thanks @1st1. A few small suggestions.

**Comment by 1st1** [comment]
@willingc Carol, I decided to introduce *awaitables* and use them to simplify (hopefully!) the docs. Please take another look.

**Review by willingc — APPROVED** [review]
I love it! I think awaitables will simplify understanding a bit. Then, we can work toward high level guidance on what is Task, Future, Coroutine are and when to consider using. :shipit:

**Comment by willingc** [comment]
Great suggestion @elprans re: *fs sequence 👍

**Pull request merged**